src/main/java/arb/model/project/Price.java

```diff
@@ -0,0 +1,61 @@
+package arb.model.project;
+
+import static arb.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Project's price in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPrice(String)}
+ */
+public class Price implements Comparable<Price> {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Price must be in a recognisable format, e.g. 4,098.09 or 4098.09 or 380";
+
+    public static final String VALIDATION_REGEX = "(([1-9]\\d{0,2}(,\\d{3})*)|(([1-9]\\d*)?\\d))(\\.\\d\\d)?$";
+    private double price;
+
+    public String fullPrice;
+
+    /**
+     * Constructs a {@code Price}.
+     * @param price A valid price.
+     */
+    public Price(String price) {
+        requireNonNull(price);
+        checkArgument(isValidPrice(price), MESSAGE_CONSTRAINTS);
+        fullPrice = price;
```

Comment:
After validating the `price [String]` you can just do `this.price = Double.parseDouble(price);` to parse it to a double and store it that way instead.

Reply:
noted with thanks

Reply:
Then to turn it to a string to store into storage, you can use `Double.toString()`